.github/workflows/ci.yml

```diff
@@ -0,0 +1,95 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - 'master'
+
+
+jobs:
+
+  lint:
+    name: Linter
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+    - name: Cache pip
+      uses: actions/cache@v2.1.6
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/dev-requirements.txt') }}
+    - name: Install libenchant
+      run: sudo apt-get install libenchant-dev
+    - name: Install dependencies
+      run: |
+        make install
+    - name: Run linters
+      run: |
+        make lint
+  
+  test:
+    name: Test
+    needs: lint
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+      redis:
+        image: redis
+        # Set health checks to wait until redis has started
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 6379:6379
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+    - name: Cache pip
+      uses: actions/cache@v2.1.6
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/dev-requirements.txt') }}
+    - name: Install libenchant
+      run: sudo apt-get install libenchant-dev
+    - name: Install dependencies
+      run: |
+        make install
```

Comment:
Also, are all these steps needed? The appear to be the same steps in lint, and this depends on lint, so I'd assume this was already done.

Reply:
I think the checkout and python steps are needed, but I am not sure about the dependency steps.

Reply:
Since we cache the dependencies during the lint job, I think it's sufficient to only include the cache step in the test job and remove the dependency steps.

Reply:
cache only caches wheels, they still need to be installed

Reply:
Thank you for the clarification.